esmtools/prediction.py

```diff
@@ -609,9 +604,78 @@ def PM_compute(ds, control, metric=_pearson_r, comparison=_m2m, anomaly=False,
         raise ValueError('specify metric argument')
 
 
+def DP_compute_skill(dp, reference, nlags=None):
+    """
+    Compute a predictability skill score for a reconstruction framework dataset.
+    
+    This is currently set up to just run a simple correlation (i.e., ACC) but
+    other metrics should be implemented if needed in the future (as in PM_compute)
+
+    Note that if reference is the reconstruction, the output correlation coefficients
+    are for potential predictability. If the reference is observations, the ouput
+    correlation coefficients are actual skill.
+
+    Parameters
+    ----------
+    dp : xarray object
+        Expected to follow package conventions (and should be an ensemble mean)
+        `ensemble` : dim of initialization dates
+        `time` : dim of lead years from those initializations
+        Additional dims can be lat, lon, depth, etc. but should not be individual
+        members.
+    reference : xarray object
+        Reconstruction or observations over same time period
+    nlags : int (default length of `time` dim)
```

Comment:
really like the idea of saving computation time here, when nlags=2

Reply:
I think this isn't relevant now that I have made changes to have a compute function for reconstruction systems.